testsuite/tests/typing-extensions/cast.ml

```diff
@@ -94,10 +94,10 @@ let c3: castable = pop_castable ();;
 
 (* We can also downcast these values to foos and bars *)
 
-let f1: foo = c1#cast (Class Foo);; (* Ok *)
-let f2: foo = c2#cast (Class Foo);; (* Ok *)
-let f3: foo = c3#cast (Class Foo);; (* Ok *)
+let f1: foo = c1#cast (Class Foo) (* Ok *);;
+let f2: foo = c2#cast (Class Foo) (* Ok *);;
+let f3: foo = c3#cast (Class Foo) (* Ok *);;
```

Comment:
What happens if you leave the `;;` in their previous place there? (Does this change has a backward-compatibility impact we should worry about?)

Reply:
If you don't move the `;;` (or at least add another one at the end of the line) the test will fail because the toplevel is waiting for more input. I'm not too worried about the backward-compatibility impact, unless some users have been using the toplevel for batch processing. For interactive use, I don't think many people type comments, and the ones who do will notice the problem immediately and type `;;`.

Reply:
Concretely if I have a `test.ml` file with contents

```
let x = 1;; (* one *)
let y = 2;; (* two *)
```

then both `ocaml test.ml` and `cat test.ml | ocaml` will process both phrases today. (The former in a silent way that is equivalent to `cat test.ml | ocaml -stdin` I believe.)  Has this changed in any way?

Reply:
Yes, and I believe this will only impact the testsuite of OCaml itself.

Reply:
Does your "yes" mean that the behavior have changed? If so, how did it change? (Is the last phrase skipped?)

Reply:
Yes, the behavior has changed. The prompts won't be in the same places, and the last group of phrases will not be executed.

last group of phrases: everything between the last double-semicolon-at-end-of-line and the end of the file.

I'm now thinking of an alternate implementation that would let us properly ignore comments and solve this problem.